aws-api-appsync/src/main/java/com/amplifyframework/datastore/appsync/SerializedModelAdapter.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.datastore.appsync;
+
+import com.amplifyframework.core.model.ModelField;
+import com.amplifyframework.core.model.ModelSchema;
+import com.amplifyframework.util.GsonObjectConverter;
+
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import com.google.gson.JsonPrimitive;
+import com.google.gson.JsonSerializationContext;
+import com.google.gson.JsonSerializer;
+
+import java.lang.reflect.Type;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Deserializer for SerializedModel. Helpful to deserialize from the graphql response.
+ */
+public final class SerializedModelAdapter
+        implements JsonDeserializer<SerializedModel>, JsonSerializer<SerializedModel> {
+    private SerializedModelAdapter() {}
+
+    /**
+     * Registers an adapter with a Gson builder.
+     * @param builder A gson builder
+     */
+    public static void register(GsonBuilder builder) {
+        builder.registerTypeAdapter(SerializedModel.class, new SerializedModelAdapter());
+    }
+
+    @Override
+    public JsonElement serialize(SerializedModel src, Type typeOfSrc, JsonSerializationContext context) {
+        ModelSchema schema = src.getModelSchema();
+
+        JsonObject result = new JsonObject();
+        result.add("id", context.serialize(src.getId()));
+        result.add("modelSchema", context.serialize(schema));
+
+        JsonObject serializedData = new JsonObject();
+        for (Map.Entry<String, Object> entry : src.getSerializedData().entrySet()) {
+            if (entry.getValue() instanceof SerializedModel) {
+                SerializedModel hpm = (SerializedModel) entry.getValue();
```

Comment:
Maybe rename `hpm` to `serializedModel`?

Reply:
Yes! This is a leftover. Will update, thanks.